pkg/util/kube/kube.go

```diff
@@ -173,16 +184,66 @@ func EnsureClusterRequirements(ctx context.Context, kubeFactory kube.Factory, na
 		return fmt.Errorf("%s: %v", requirementsValidationErrorMessage, specificErrorMessages)
 	}
 
-	err = runNetworkTest(ctx, kubeFactory, contextUrl)
+	err = runNetworkTest(ctx, kubeFactory, runtimeInstallOptions.ContextUrl)
 	if err != nil {
 		return fmt.Errorf("cluster network tests failed: %w ", err)
 	}
 
 	log.G(ctx).Info("Network test finished successfully")
 
+	if runtimeInstallOptions.AccessMode == platmodel.AccessModeTunnel {
+		err = runTCPConnectionTest(ctx, &runtimeInstallOptions)
+		if err != nil {
+			return fmt.Errorf("cluster TCP connection tests failed: %w ", err)
+		}
+
+		log.G(ctx).Info("TCP connection test finished successfully")
+	}
 	return nil
 }
 
+func runTCPConnectionTest(ctx context.Context, runtimeInstallOptions *RuntimeInstallOptions) error {
```

Comment:
can this func, or maybe some of this func, be merged with the func that tests http network?
or have you already extracted all the similar code from both, as much as it makes sense?

Reply:
I think yes. I dont see what I can move to function except handlePodState